web/middleware.py

```
@@ -141,3 +142,56 @@ def __call__(self, request):
+            allowed_urls = ["take_quiz"]
+
+            # If navigating to a non-allowed URL while quiz is active
+            if current_url_name not in allowed_urls:
+                # Get the active quiz
+                try:
+                    user_quiz = UserQuiz.objects.get(id=active_quiz_id, user=request.user, completed=False)
+                    # Mark quiz as completed
+                    try:
+                        user_quiz.complete_quiz()
+                        user_quiz.save()
+                    except Exception as e:
```

Ruff (0.8.2)

Do not catch blind exception: `Exception`

(BLE001)